app/controllers/buildings_controller.ts

```diff
@@ -1,70 +1,83 @@
 import Building from '#models/building'
-import Campus from '#models/campus'
+import { showValidator, indexValidator, byCampusValidator } from '#validators/building'
 import { HttpContext } from '@adonisjs/core/http'
 
 export default class BuildingsController {
   /**
    * Display a list of resource
    */
-  async index({ response }: HttpContext) {
-    try {
-      const buildings = await Building.all()
-      if (buildings.length === 0) {
-        return response.status(400).json({ message: 'No buildings in database' })
-      }
-      return response.status(200).json(buildings)
-    } catch (error) {
-      return response
-        .status(500)
-        .json({ message: 'Failed to fetch buildings', error: error.message })
+  async index({ request, response }: HttpContext) {
+    const { page, limit, includeCampus } = await request.validateUsing(indexValidator)
+
+    const buildingsQuery = Building.query()
+    const message = 'List of buildings'
+
+    if (includeCampus) {
+      buildingsQuery.preload('campus')
+    }
+
+    if (page) {
+      const buildings = await buildingsQuery.paginate(page, limit || 10)
```

Comment:
`||` works okay in this case, but in general when you are checking for undefined it is better to use `??` operator